code/controllers/subsystem/garbage.dm

```diff
@@ -169,16 +176,21 @@ SUBSYSTEM_DEF(garbage)
 		if(GCd_at_time > cut_off_time)
 			break // Everything else is newer, skip them
 		count++
+
+#ifdef EXPERIMENT_515_QDEL_HARD_REFERENCE
+		var/datum/D = L[2]
+#else
 		var/refID = L[2]
 		var/datum/D
 		D = locate(refID)
+#endif
 
-		if (!D || D.gc_destroyed != GCd_at_time) // So if something else coincidently gets the same ref, it's not deleted by mistake
+		if (IS_DELETED(D, GCd_at_time)) // So if something else coincidently gets the same ref, it's not deleted by mistake
 			++gcedlasttick
 			++totalgcs
 			pass_counts[level]++
 			#ifdef REFERENCE_TRACKING
-			reference_find_on_fail -= refID //It's deleted we don't care anymore.
+			reference_find_on_fail -= text_ref(D) //It's deleted we don't care anymore.
```

Comment:
will add overhead to unit tests btw

Reply:
515 tests seem to run faster than the 514 ones :P

Reply:
~~I meant the 514 ones~~